AwsCryptographicMaterialProviders/dafny/AwsCryptographyKeyStore/src/DefaultKeyStorageInterface.dfy

```diff
@@ -585,78 +585,83 @@ module DefaultKeyStorageInterface {
+      ];
+      var commitmentReq := DDB.GetItemInput(
+        Key := commitmentKey,
+        TableName := ddbTableName,
+        ConsistentRead :=  Some(true)
+      );
+      var commitmentRes? := ddbClient.GetItem(commitmentReq);
+      var commitmentRes :- commitmentRes?
       .MapFailure((e: DDB.Error) =>
                     wrapDdbException(
                       e:=e,
                       storageOperation:="GetMutation",
-                      ddbOperation:="TransactGetItems",
+                      ddbOperation:="GetItem",
                       identifier:=input.Identifier,
                       tableName:=ddbTableName));
 
         // SDKs/Smithy-Dafny/Custom Implementations of Storage MAY respond with None or an Empty Map.
         // .NET returns an empty map, Java returns None.
       :- Need(
-        ddbResponse.Responses.Some? && (2 == |ddbResponse.Responses.value|),
+        commitmentRes.Item.Some? && (0 < |commitmentRes.Item.value|),
+        Types.KeyStorageException(
+          message:=
+            "GetMutation's GetItem for the Mutation Commitment returned an empty result. "
+            + "Branch Key ID: " + input.Identifier
+            + "\tTable Name: " + ddbTableName));
+
+      // Get Mutation Index
+      var indexKey: DDB.Key := map[
+        Structure.BRANCH_KEY_IDENTIFIER_FIELD := DDB.AttributeValue.S(input.Identifier),
+        Structure.TYPE_FIELD := DDB.AttributeValue.S(Structure.MUTATION_INDEX_TYPE)
+      ];
+      var indexReq := DDB.GetItemInput(
+        Key := indexKey,
+        TableName := ddbTableName,
+        ConsistentRead :=  Some(true)
+      );
+      var indexRes? := ddbClient.GetItem(indexReq);
+      var indexRes :- indexRes?
+      .MapFailure((e: DDB.Error) =>
+                    wrapDdbException(
+                      e:=e,
+                      storageOperation:="GetMutation",
+                      ddbOperation:="GetItem",
+                      identifier:=input.Identifier,
+                      tableName:=ddbTableName));
+      // SDKs/Smithy-Dafny/Custom Implementations of Storage MAY respond with None or an Empty Map.
+      // .NET returns an empty map, Java returns None.
+      :- Need(
+        indexRes.Item.Some? && (0 < |indexRes.Item.value|),
         Types.KeyStorageException(
           message:=
-            "GetMutation: No items returned. "
+            "GetMutation's GetItem for the Mutation Index returned an empty result. "
             + "Branch Key ID: " + input.Identifier
             + "\tTable Name: " + ddbTableName));
 
       /** Process sensical DDB Response */
-      var lockCanidate := ddbResponse.Responses.value[0].Item;
+      ghost var lockCanidate := commitmentRes.Item;
       var lockItem: Option<Types.MutationCommitment> :-
-        MutationCommitmentFromOptionalItem(lockCanidate, input.Identifier, ddbTableName);
+        MutationCommitmentFromOptionalItem(commitmentRes.Item, input.Identifier, ddbTableName);
```

Comment:
You `Need` `commitmentRes.Item.Some? && (0 < |commitmentRes.Item.value|)` above, so you can use `StorageHelpers.MutationCommitmentFromItem` and avoid the wrapping.

Reply:
FYI: this method can be called and succeed with an empty result (no Mutation in-flight, for Describe Mutation).
I do not think we need to un-wrap these...